components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/connector/RDBMSConnector.java

```diff
@@ -149,8 +150,8 @@ public long getTransactionCountOfMonth(int year, int monthNumber) throws Transac
         try (Connection dbConnection = getConnection();
              PreparedStatement prepStmt = dbConnection.prepareStatement(
                      TransactionQueryHelper.GET_TRAN_COUNT_DATA_FOR_A_TIME_PERIOD)) {
-            prepStmt.setString(1, startDate);
-            prepStmt.setString(2, endDate);
+            prepStmt.setDate(1,  new java.sql.Date(ConverterUtil.convertToDate(startDate).getTime()));
+            prepStmt.setDate(2, new java.sql.Date(ConverterUtil.convertToDate(endDate).getTime()));
```

Comment:
shall we add this new java.sql.Date(ConverterUtil.convertToDate(endDate).getTime()) to a function and re use ( a signle line function .. ) use it in line 238 as well.

Reply:
Fixed